nat-lab/tests/test_batching.py

```diff
@@ -129,61 +188,63 @@ async def test_batching_star(
-        for name in container_names:
-            task = asyncio.create_task(
-                capture_traffic(
-                    name,
-                    capture_duration,
-                )
-            )
-            pcap_capture_tasks.append(task)
-
         # misalign the peers by sleeping some before starting each node again
-        for client, node in cnodes:
-            await asyncio.sleep(misalign_sleep_amount)
+        async def start_node_manually(client, node, sleep_min: int, sleep_max: int):
+            await asyncio.sleep(random.randint(sleep_min, sleep_max))
             await client.simple_start()
             await client.set_meshmap(env.api.get_meshmap(node.id))
 
-        await asyncio.gather(*(
-            client.wait_for_state_peer(
-                peer_node.public_key, [State.Connected], [PathType.Direct]
+        await asyncio.gather(*[
+            start_node_manually(
+                client, node, misalign_sleep_range[0], misalign_sleep_range[1]
             )
-            for client, peer_node in product(env.clients, env.nodes)
-            if not client.is_node(peer_node)
-        ))
+            for client, node in cnodes
+        ])
 
-        pcap_paths = await asyncio.gather(*pcap_capture_tasks)
+        # pylint: disable
+        pyro5_ports = [
+            int(port)
+            for port in list(set([client.get_proxy_port() for client in env.clients]))
+        ]
 
-        filters = [
-            None,
-            ("icmpv6echorequest", lambda p: p.haslayer(ICMPv6EchoRequest)),
+        allow_pcap_filters = [
             (
-                "ipv46 udp",
+                "IP46 + No Pyro5 traffic",
                 lambda p: (
-                    (p.haslayer(IP) or p.haslayer(IPv6)) if p.haslayer(UDP) else True
+                    (p.haslayer(UDP) or p.haslayer(TCP))
+                    and p.sport not in pyro5_ports
+                    and p.dport not in pyro5_ports
                 ),
             ),
         ]
 
+        pcap_capture_tasks = []
+        for name in container_names:
+            pcap_task = asyncio.create_task(
+                capture_traffic(
+                    name,
+                    capture_duration,
+                )
+            )
+            pcap_capture_tasks.append(pcap_task)
+
+        pcap_paths = await asyncio.gather(*pcap_capture_tasks)
+
         for container, pcap_path in zip(container_names, pcap_paths):
-            for filt in filters:
+            for filt in allow_pcap_filters:
                 filter_name = filt[0] if filt else "none"
                 hs = generate_histogram_from_pcap(
                     pcap_path, capture_duration, filt[1] if filt else None
                 )
                 title = f"{container}-filter({filter_name})"
-                save_histogram(title, hs, "logs", max_height=12)
-
-        for p in pcap_paths:
-            os.unlink(p)
+                print_histogram(title, hs, max_height=12)
```

Comment:
What are the test conditions here? or is this only intended for manual human inspection?
And do you think about having some rudimentary check like that histogram is not empty?

Reply:
For now the plan is to just observe it and later on bolt on some assertions. This means manual inspection after nightly and/or manual runs :) 

This is not ideal and we could already do some asserts at this point but the CI runs are pretty slow so at this point I am satisfied by just having histograms somewhere for observation :)